### PR TITLE
Add checkDirExists endpoint + tests

### LINKED ISSUE #13: Check if directory exists
An endpoint that checks if a specified directory exists.

- Verb: `GET`
- Path: `/dir/exists`

Query parameters:
| Name | Value | Default value | Description |
| --- | --- | --- | --- |
| `path` | string | **required** | A path to a directory which existence should be checked |

Responses:
- `204 No Content` - when the directory exists we can simply return `204` to say "found it". No need to send any additional data.
- `404 Not Found` - when couldn't find the directory or if the path is pointing to a file.
- `400 Bad Request` - when the path is invalid.

### PR DESCRIPTION
**Pull request by lansolo99**
This PR closes #13 
I mostly created derived logic from checkIfFileExists.
I also created the relative tests. I was able to find how to test the status 400 properly, it fallbacks to 404.

I noticed a lot of test errors when executing ```bun test```
My coverage is 92%.

### CODE REVIEW

**Comment by Xkonti** [comment]
Tests are suggesting issue with the positive test:
```
1 tests failed:
(fail) should return HTTP Status 204 if the directory is found [0.95ms]
```

```
Check failure on line 37 in src/endpoints/dir/checkDirExists.test.ts

GitHub Actions
/ check-pr
error: expect(received).toBe(expected)
Expected: 204
Received: 404
```

**Comment by lansolo99** [comment]
@Xkonti  Fixed the status 204 test by adding proper filter to the tested directories terms.

File: src/endpoints/dir/checkDirExists.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    // Verify that the path is valid
+    let relativePath = query.path ? (query.path as string) : null;
+    if (!isPathValid(relativePath)) {
+      set.status = 400;
+      return provideValidPathDirMsg;
+    }
+    relativePath = relativePath as string;
+
+    // Get the full path to the directory
+    const dirPath = join(getConfig().dataDir, relativePath);
```
Comment (Xkonti) [inline src/endpoints/dir/checkDirExists.ts]: You can get the relative path and absolute path at the same time using `validateRelativePath`. This way we will avoid same checking logic in multiple spots:

```cs
// Performs validation of the `query.path`
const dirPathValidationResult = validateRelativePath(query.path);
if (dirPathValidationResult.isErr()) {
  set.status = 400;
  return provideValidPathDirMsg;
}
// Create variable to hold relative and absolute path
const dirPath = sourcePathValidationResult.value;

// Using the relative and absolute path:
let something = dirPath.relativePath;
let somethingElse = dirPath.absolutePath;
```

**Review by Xkonti — CHANGES_REQUESTED** [review]
Looks great. There's just one last thing before I can merge it in. Thank you!

File: src/endpoints/dir/checkDirExists.ts (hunk 1 above)
Comment (lansolo99) [reply src/endpoints/dir/checkDirExists.ts]: I reused it. Though the absolutePath doesn't seems necessary for this endpoint, I just used the relativePath from the value.

Comment (Xkonti) [reply src/endpoints/dir/checkDirExists.ts]: Looks like you're using the absolute path in line 19:

```cs
// Get the full path to the directory
const dirPath = join(getConfig().dataDir, relativePath);
```

Here's whot it looks like in `validateRealativePath`:
https://github.com/Xkonti/file-api/blob/bd3b6fbfd75e3aea70dae70f8f9d1470f91a2869/src/utils/pathUtils.ts#L27-L30

**Comment by lansolo99** [comment]
@Xkonti I didn't notice it was the same piece of logic, my bad. So I only use the already returned absolutePath this time.

**Review by Xkonti — APPROVED** [review]
Looks awesome. Thanks!

**Pull request merged**